src/status_im/contexts/wallet/send/events.cljs

```diff
@@ -50,7 +50,7 @@
    {:db (-> db
             (assoc-in [:wallet :ui :send :send-account-address] address)
             (update-in [:wallet :ui :send] dissoc :to-address))
-    :fx [[:navigate-to-within-stack [:wallet-select-asset stack-id]]]}))
+    :fx [[:dispatch [:navigate-to-within-stack [:wallet-select-asset stack-id]]]]}))
```

Comment:
HI @ulisesmac, this is because we need to update `:modal-stack-ids` in app db, not just execute the push in `react-native-navigation`, also checked and this is how we are dispatching navigation events through the app in events. Without this, sub `:navigation/current-screen-id` would return wrong data.

Reply:
Really interesting.
It seems we are just delaying the evaluation of the navigation to let other re-frame updates successfully happen.

Too much to ask for :sweat_smile:  but if you are free at some point, could you please check (it's a long discussion, it's ok if you can't):
https://github.com/status-im/status-mobile/pull/18532#issuecomment-1909279658

Basically, I'm delaying the navigation to let re-frame properly update it's state.

Reply:
It's super interesting, indeed. We are updating the db also in our main `:navigate-to` event. Here https://github.com/status-im/status-mobile/blob/01d1ff0cfb0a863d294937dea91b892237aa7c49/src/status_im/contexts/shell/jump_to/events.cljs#L121
 and here https://github.com/status-im/status-mobile/blob/972a88e7ccc9fd5fc697caa42919079a4da73859/src/status_im/navigation/events.cljs#L19

I've read the discussion, really interesting data shared there. If I understood correctly, the db is updated properly, but the problems appear when using subscriptions in the mount phase. What if you dispatch the event without delay instead of using effect? That's how navigation events are used though the app, and may have caused the issue in that PR, not sure.

Reply:
> What if you dispatch the event without delay instead of using effect?

It seems the `:dispatch` is not enough, we need even more delay, at least 1ms and sometimes it is still happening.